docs/references/http-gateway-protocol-spec.md

```diff
@@ -0,0 +1,471 @@
+# The HTTP Gateway Protocol Specification
+
+## Introduction
+
+The HTTP Gateway Protocol is an extension of the Internet Computer Protocol that allows conventional HTTP clients to interact with the Internet Computer network. This is important for software such as web browsers to be able to fetch and render client-side canister code, including HTML, CSS, and JavaScript as well as other static assets such as images or videos. The HTTP Gateway does this by translating between standard HTTP requests and requests that the Internet Computer Protocol will understand.
+
+Such a gateway could be a stand-alone proxy, it could be implemented in web browsers (natively, via a plugin or a service worker) or in other ways. This document describes the interface and semantics of this protocol independent of a concrete Gateway so that all Gateway implementations can be compatible.
+
+## Overview
+
+An HTTP request by an HTTP client is handled by these steps:
```

Comment:
Just an idea, not sure it makes sense: Highlight who is doing the step by e.g., adding "Client", "Gateway", "Canister" to the step.

In addition, maybe we can add a step 0. that is "Intercept the HTTP request from the HTTP client" to have a counterpart for step 8.

Reply:
This makes sense, I'll add the "actor" in for each step

Reply:
Wait I'm confused, because you suggested to remove "the gateway" in the following steps :D could you clarify what you mean?

Reply:
I suggested to remove the gateway in the following steps because the steps were not consistent. The first steps were written from the point of view of the gateway (e.g. "resolve the canister ID") and then they were written from the point of view as a bystander (e.g., "if applicable, the Gateway fetches"). I then thought we should make clear "who" is doing the step as step 4 doesn't really fit in with the rest. 

So I suggest, we either switch to something like 4: e.g., "The gateway resolves the canister ID..." or something like "Gateway: Resolve the canister ID..." and "Canister: Handle the request..."

Reply:
Understood, I'll update it